mock/service/controller.go

```diff
@@ -326,6 +326,41 @@ func (s *service) ValidateVolumeCapabilities(
 	}, nil
 }
 
+func (s *service) ControllerGetVolume(
+	ctx context.Context,
+	req *csi.ControllerGetVolumeRequest) (
+	*csi.ControllerGetVolumeResponse, error) {
+
+	if hookVal, hookMsg := s.execHook("GetVolumeStart"); hookVal != codes.OK {
+		return nil, status.Errorf(hookVal, hookMsg)
+	}
+
+	resp := &csi.ControllerGetVolumeResponse{
+		Status: &csi.ControllerGetVolumeResponse_VolumeStatus{
+			VolumeCondition: &csi.VolumeCondition{},
+		},
+	}
+	i, v := s.findVolByID(ctx, req.VolumeId)
+	if i < 0 {
+		resp.Status.VolumeCondition.Abnormal = true
+		resp.Status.VolumeCondition.Message = "volume not found"
+		return resp, status.Error(codes.NotFound, req.VolumeId)
+	}
```

Comment:
Can you add a case where volume is found but condition is Abnormal=true?

Reply:
For the mock test, I think we can divide it into 2 parts to do
1. Add basic mock test functions for testing if volume-healthy-monitor can work(Due to the initial implementation PR wasn’t merged, it is hard to do further works)
2. Add more unit, integration, e2e test for testing volume-healthy-monitor can work correctly on positive and negative cases

The reasons I want to do a testing task like above are below
1. The initial PR blocks many works and it isn’t perfect. We can ensure it works normally and improves it
2. After initial PR was merged. It is better for us to let more contributors contribute this project.

Reply:
Sure